### PR TITLE
feat: Support use blob to createComponent

### PR DESCRIPTION
**Pull request by WaylonFtrack**

### CODE REVIEW

**Comment by jimmycallin** [comment]
Hi @WaylonFtrack! Would you mind looking into getting the tests for file uploading to work (they are currently skipped), and add an additional test for blobs? Thanks!

**Review by jimmycallin — CHANGES_REQUESTED** [review]
Hi @WaylonFtrack! Would you mind looking into getting the tests for file uploading to work (they are currently skipped), and add an additional test for blobs? Thanks!

**Comment by lucaas** [comment]
Is the goal to add support for browsers not supporting the [File constructor](https://developer.mozilla.org/en-US/docs/Web/API/File), or what is the motivation for the change?

**Comment by WaylonFtrack** [comment]
I was trying to upload canvas image from [HTMLCanvasElement.toBlob](https://devdocs.io/dom/htmlcanvaselement/toblob). Since file constructor is additional wrapper, and the changes to directly support `Blob` are minimal and compatible with `File`, I submitted this PR.

File: source/session.js [hunk 1]
```diff
@@ -825,9 +829,13 @@ export class Session {
     let headers;
 
     const updateOnProgressCallback = (oEvent) => {
+      let progress = 0;
```
Comment (WaylonFtrack) [inline source/session.js:832]: jsdom won't set request headers before `request`, cause `onProgress` never be called in test enviroment.

**Review by jimmycallin — APPROVED** [review]
I'm happy with this, but would like @lucaas to approve as well! Good job with the tests as well!

**Comment by lucaas** [comment]
> I was trying to upload canvas image from [HTMLCanvasElement.toBlob](https://devdocs.io/dom/htmlcanvaselement/toblob). Since file constructor is additional wrapper, and the changes to directly support `Blob` are minimal and compatible with `File`, I submitted this PR.

I see. In the review player, we also use `canvas.toBlob`, but duck-type it as a File by [just setting blob.name](https://github.com/ftrackhq/frontend/blob/35ff23182d510cbb16329540415152036b9fa7b7/packages/overview/source/lib/cso/export_annotation.js#L24-L29). This is a cleaner solution, so happy to accept it.

**Review by lucaas — APPROVED** [review]

**Pull request merged**